torch/_inductor/ir.py

```diff
@@ -4542,6 +4544,52 @@ def __init__(
         self.kwargs = {} if kwargs is None else kwargs
         V.graph.warn_fallback(self.python_kernel_name)
 
+        # args that are aliased
+        self.alias_names = []
+        # args that are mutated AND returned from the op
+        self.mutation_names = []
+
+        if isinstance(self.op_overload, torch._ops.HigherOrderOperator):
+            # We assume here that HOPs with FallbackKernel are functional.
+            # This may not always be true! HOPs must individually opt-in to
+            # FallbackKernel, so please check this if you opt-in.
+            return
```

Comment:
Can we check this assumption somehow?

Reply:
No, there's no notion of a schema for HOPs. Though we generally tell people that they must be functional.